# Higher Taxonomic Classification of Uncatalogued Species
This repository contains an implementation of a hierarchical metagenome fragment classifier used to test the hypothesis that a hierarchical arrangement of support vector machines can be used to classify genome fragments from species not seen during training at higher taxonomic levels including Domain, Phylum and Genus.  A key feature of this model architecture is the ability to determine on a per-fragment basis at what taxonomic level a classification can be confidently assigned (this uses the probability modelling feature of scikit-learn SVMs).  This model is similar to the viral metagenome classifier [CHEER](https://www.sciencedirect.com/science/article/pii/S1046202319302683), but the individual SVM models are trained on tetranucleotide frequency vectors from prokaryotic metagenome fragments, similarly to [Perry and Beiko](https://pubmed.ncbi.nlm.nih.gov/20333228/).  The pipeline code can be found within RHamilton_final_pipeline.ipynb and the report can be found in RHamilton_final_report.pdf.

This was created as the final project for 410.734: Practical Introduction to Metagenomics at Johns Hopkins University.  Note that this architecture was only trained on a small subset of the prokaryotic taxonomy (2 Archaeal and 2 Bacterial phyla) to ensure that something could actually be produced within the class time frame.
